workers/unity/Packages/com.improbable.gdk.core/Worker/ConnectionUtils.cs

```diff
@@ -0,0 +1,135 @@
+using System.Threading.Tasks;
+using Improbable.Worker.CInterop;
+using AlphaLocator = Improbable.Worker.CInterop.Alpha.Locator;
+
+namespace Improbable.Gdk.Core
+{
+    public static class ConnectionUtils
+    {
+        /// <summary>
+        ///     Connects to the SpatialOS Runtime via the Receptionist service and creates a <see cref="Connection" /> object
+        ///     asynchronously.
+        /// </summary>
+        /// <param name="config">
+        ///     The <see cref="ReceptionistConfig" /> object stores the configuration needed to connect via the
+        ///     Receptionist Service.
+        /// </param>
+        /// <param name="connectionParameters">The <see cref="ConnectionParameters" /> storing </param>
+        /// <returns>
+        ///     A <see cref="Task{TResult}" /> to run this method asynchronously and retrieve the created
+        ///     <see cref="Connection" /> object upon connecting successfully.
+        /// </returns>
+        public static async Task<Connection> CreateConnectionAsync(
+            ReceptionistConfig config,
+            ConnectionParameters connectionParameters)
+        {
+            using (var connectionFuture =
+                Connection.ConnectAsync(config.ReceptionistHost, config.ReceptionistPort, config.WorkerId,
+                    connectionParameters))
+            {
+                return await TryToConnectAsync(connectionFuture).ConfigureAwait(false);
+            }
+        }
+
+        /// <summary>
+        ///     Connects to the SpatialOS Runtime via the Locator service and creates a <see cref="Connection" /> object
+        ///     asynchronously.
+        /// </summary>
+        /// <param name="parameters">
+        ///     The <see cref="LocatorConfig" /> object stores the configuration needed to connect via the
+        ///     Locator Service.
+        /// </param>
+        /// <param name="connectionParameters">The <see cref="ConnectionParameters" /> storing </param>
+        /// <returns>
+        ///     A <see cref="Task{TResult}" /> to run this method asynchronously and retrieve the created
+        ///     <see cref="Connection" /> object upon connecting successfully.
+        /// </returns>
+        public static async Task<Connection> CreateConnectionAsync(
+            LocatorConfig parameters,
+            ConnectionParameters connectionParameters)
+        {
+            using (var locator = new Locator(parameters.LocatorHost, parameters.LocatorParameters))
+            {
+                var deploymentList = await GetDeploymentList(locator);
+
+                var deploymentName = parameters.DeploymentListCallback(deploymentList);
```

Comment:
Hmmm - we should allow for the possibility that this callback can be asynchronous. 

I.e - user displays to the client and they select a deployment

Reply:
we probably don't want that. the login token that is provided by the launcher will most likely time out by the time the user selects a deployment. 
If users want to do that, they need to add quite a bit of additional logic and will most likely need to use the alpha locator together with dev auth flow or their own auth server.

Reply:
This was just rearranging what was there. If we can't select a deployment without a token and we can't keep a token after selecting a deployment that is a problem though.

Reply:
I agree, I had many discussions about this...